### PR TITLE
Domain block improvements

### PR DESCRIPTION
**Pull request by Gargron**
* Make file attachment on MediaAttachment optional
* Create MediaAttachment but without actual file download when domain is blocked with reject_media set to true
* Clean up old media files when creating a new domain block with reject_media set to true
* Return remote_url in media attachments API if local file is not present
* Undo domain block action in admin UI
* Ability to enable reject_media from admin UI

### CODE REVIEW

**Comment by ashfurrow** [comment]
This is great, thank you!

**Comment by mjankowski** [comment]
Domain block stuff looks good.

Is the view spec stuff (fix?) here on purpose?

**Comment by Gargron** [comment]
@mjankowski I kinda had to add it here since I wanted the tests to pass. I didn't notice that the tests were still broken when merging the other thing, sadly - I just looked through the code and it looked good to me.

**Comment by mjankowski** [comment]
Cool.

I just submitted this for view spec fix only: https://github.com/tootsuite/mastodon/pull/1888

If that looks good, maybe merge that one first and then rebase this with just the domain stuff?

File: app/services/block_domain_service.rb [hunk 1]
```diff
@@ -3,12 +3,34 @@
 class BlockDomainService < BaseService
   def call(domain_block)
     if domain_block.silence?
-      Account.where(domain: domain_block.domain).update_all(silenced: true)
```
Comment (mjankowski) [inline app/services/block_domain_service.rb]: I like the change to move this to smaller methods.  Possible further refactor...

- Set `@domain_block` accessor right in call, and then call a method like `process_block` or something, and do nothing else in `call`
- `process_block` would have the if/else logic around silence vs suspend
- Add a method like `blocked_domain` which is `domain_block.domain` and replace a few of those
- Add something like `domain_accounts` which does the Account.where(...)` that's repeated a few times here

File: config/locales/en.yml [hunk 2]
```diff
@@ -90,8 +92,22 @@ en:
           silence: Silence
           suspend: Suspend
         title: New domain block
+      reject_media: Reject media files
+      reject_media_hint: Removes locally stored media files and refuses to download any in the future. Irrelevant for suspensions
```
Comment (mjankowski) [inline config/locales/en.yml]: Period at end of sentence?

**Review by mjankowski — APPROVED** [review]
Left a note about a possible refactor, but looks good overall.

**Comment by Angristan** [comment]
It would be great to have the "Reject media files" option without blocking or suspending the instance.

**Comment by Angristan** [comment]
Another idea : proxying media files without caching them. Otherwise it will be a mess with CSP.

**Comment by Gargron** [comment]
@Angristan How to deal with WebM/MP4? Proxying on the fly would be one heck of a load.

**Comment by Angristan** [comment]
True... About that, we should at least have the ability of choosing the delay of caching. It's a huge waste of space, in a few feeks/month, every instance will have dozens of GBs of medias stored that aren't even accessed anymore. :/

**Pull request merged**